src/java.base/share/classes/jdk/internal/vm/ThreadDumper.java

```diff
@@ -205,7 +205,10 @@ private static boolean dumpThread(Thread thread, TextWriter writer) {
                 // park blocker
                 Object parkBlocker = snapshot.parkBlocker();
                 if (parkBlocker != null) {
-                    writer.println("    - parking to wait for " + decorateObject(parkBlocker));
+                    String suffix = (snapshot.parkBlockerOwner() != null)
```

Comment:
This was `if (snapshot.parkBlockerOwner() instanceof Thread owner)` in the original prototype. It avoids the null check and allow L209 to use owner.threadId(). It also is consistent with the other use of pattern matching with instanceof in this code.

Reply:
Fixed (and for json version too)